### ISSUE TITLE
aio: heading anchor accessibility

### ISSUE THREAD

**Issue by petebacondarwin** [angular/angular#16854]
The `header-link` anchor tags used to deep link headings are poorly constructed:

The anchors have `aria-hidden="true"` but they still have hrefs, causing what we call "ghost controls" with no accessibility information when focused. Pair that with a hover state on the child icon element only and no focus state to make them visible, and they are pretty bad accessibility-wise.

Here's a screenshot:
![aio-header-links](https://cloud.githubusercontent.com/assets/1045233/26172761/5c128526-3afe-11e7-9f0f-def052b36ab5.png)

To fix these, I'd move the hover state to the anchor and add a focus state at the same time. Remove `aria-hidden="true"` and they should work better.

**Comment by IgorMinar** [comment]
is this still a problem?

**Comment by petebacondarwin** [comment]
This info originally came from @marcysutton. I am not 100% sure what changes would need to be made. But I can look into it this week.

**Comment by petebacondarwin** [comment]
I just ran [Axe Coconut](https://chrome.google.com/webstore/detail/axe-coconut-web-accessibi/iobddmbdndbbbfjopjdgadphaoihpojp?hl=en) on the site and it is not indicating that we have ghost controls.
